### PR TITLE
move notification handles and events metadata into api

### PR DESCRIPTION
**Pull request by wy65701436**
1, enable audit logs for notifications
2, move the handler and meatadata into API
3, use the notification middleware to send out notification

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: src/api/event/handler/chart/chart.go [hunk 1]
```diff
@@ -1,28 +1,39 @@
-package notification
```
Comment (ywk253100) [inline src/api/event/handler/chart/chart.go]: Seems all handlers under the package `src/notifier/handler/notification` are webhook related? If they are, how about moving all of then into a separated package?

Comment (wy65701436) [reply src/api/event/handler/chart/chart.go]: only http and slack are left, all the others are move into ./api/event/handlers

File: src/api/event/handler/init.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	notifier.Subscribe(event.TopicDeleteArtifact, &auditlog.Handler{})
+	notifier.Subscribe(event.TopicCreateProject, &auditlog.Handler{})
+	notifier.Subscribe(event.TopicDeleteProject, &auditlog.Handler{})
+	notifier.Subscribe(event.TopicDeleteRepository, &auditlog.Handler{})
+	notifier.Subscribe(event.TopicCreateTag, &auditlog.Handler{})
+	notifier.Subscribe(event.TopicDeleteTag, &auditlog.Handler{})
+}
+
+// InitSupportedEvent ....
+func InitSupportedEvent() {
```
Comment (ywk253100) [inline src/api/event/handler/init.go]: Move this into webhook related package

Comment (wy65701436) [reply src/api/event/handler/init.go]: move it into api

**Pull request merged**